docs/connectors/secret-references.md

```diff
@@ -0,0 +1,100 @@
+# Secret references
+
+Rather than directly supplying a secret (e.g., passwords, PEM, etc.) in the connector
+configuration as a string, it is possible to refer to (externalize) them. This mechanism
+in Feldera is called a **secret reference**.
+
+A secret reference is a string in the connector configuration JSON which takes a specific format:
+
+```
+${secret:<provider>:<identifier>}
+```
+
+It refers to an identifiable secret provided by a provider. Feldera's control plane mounts the secret
+into the pipeline. When the pipeline initializes, it will replace the secret references in the
+configuration with their values. We currently only support a single secret provider, Kubernetes.
+
+## Kubernetes
+
+### Usage
+
+```
+${secret:kubernetes:<name>/<data key>}
+```
+
+... in which:
+- `<name>` should correspond to a Kubernetes `Secret`
+- `<data key>` should correspond to a data key value (i.e., in its `data:` section)
+
+Upon provisioning the pipeline, the runner will mount the Kubernetes Secret data keys
+as files at: `/etc/feldera-secrets/<name>/<data key>`.
+
+:::info
+
+The automatic resolution of Kubernetes Secrets through mounting is Enterprise-only.
+In the Open Source edition, it is still possible to refer to secrets by carefully
+mounting them yourself out-of-band at `/etc/feldera-secrets/kubernetes/<name>/<data key>`.
```

Comment:
One of these paths contains `kubernetes` and the other does not. I don't know whether that is a mistake.

Reply:
Yep, that's a typo, will fix it. Good catch :+1: